applications/FluidDynamicsApplication/tests/compressible_navier_stokes_symbolic_generator_formulation_test.py

```diff
@@ -0,0 +1,148 @@
+import KratosMultiphysics.KratosUnittest as KratosUnitTest
+
+from KratosMultiphysics.FluidDynamicsApplication.symbolic_generation.compressible_navier_stokes \
+    .compressible_navier_stokes_symbolic_generator import CompressibleNavierStokesSymbolicGenerator
+
+import KratosMultiphysics.FluidDynamicsApplication
+
+class CompressibleNavierStokesSymbolicGeneratorFormulationTest(KratosUnitTest.TestCase):
+    def setUp(self):
+        self.files_to_remove = []
+
+    def tearDown(self):
+        for file_name in self.files_to_remove:
+            KratosMultiphysics.kratos_utilities.DeleteFileIfExisting(file_name)
+        self.files_to_remove = []
+
+    @classmethod
+    def _GetGeneratorSettings(cls, geometry):
+        """
+        Returns the Kratos Parameters for the symbolic generator
+        """
+        geometry_name = {
+            "2D3N": "triangle",
+            "2D4N": "quadrilateral",
+            "3D4N": "tetrahedron"
+        }[geometry]
+
+        parameters = KratosMultiphysics.Parameters("""
+        {
+            "geometry": "PLEASE SPECIFY A GEOMETRY",
+            "template_filename" : "PLEASE SPECIFY A TEMPLATE FILE",
+            "output_filename"   : "PLEASE SPECIFY AN OUTPUT FILE",
+            "language" : "python"
+        }""")
+
+        parameters["geometry"].SetString(geometry_name)
+        parameters["template_filename"].SetString("test_{}.py_template".format(geometry))
+        parameters["output_filename"].SetString("symbolic_test_{}.py".format(geometry))
+
+        return parameters
+
+    @classmethod
+    def _Generate(cls, geometry):
+        params = cls._GetGeneratorSettings(geometry)
+        generator = CompressibleNavierStokesSymbolicGenerator(params)
+        generator.Generate()
+        generator.Write()
+
+        return generator.output_filename
+
+    @classmethod
+    def _ImportSubTestSuite(cls, generated_file_name):
+        module_name = "compressible_symbolic_generation"
+
+        try:
+            test_module = __import__(module_name + "." + generated_file_name[:-3], fromlist=module_name)
+        except ModuleNotFoundError as err:
+            print("Failed to import generated file:", module_name)
+            raise err
+        return test_module.SubTestSuite()
+
+
+    def _RunSubTestSuite(self, sub_testsuite,print_results):
+        tests = [testname for testname in dir(sub_testsuite)
+            if callable(getattr(sub_testsuite, testname)) and testname.startswith("test_")]
+
+        for testname in tests:
+            try:
+                result, reference = getattr(sub_testsuite, testname)()
+            except Exception as e:
+                raise RuntimeError("Error in test " + testname) from e
+
+            if print_results:
+                print(result)
+
+            try:
+                self.assertVectorAlmostEqual(result, reference)
+            except AssertionError as e:
+                raise AssertionError("AssertionError in " + testname) from e
```

Comment:
This will become a one-liner if #9635 is accepted:
```
self.assertVectorAlmostEqual(result, reference, msg="Failure in sub-test "+testname)
```

The error message is also much cleaner.

Reply:
doesn't it automatically show which test failed?